app/Member.php

```diff
@@ -4,27 +4,53 @@
 
 use Carbon\Carbon;
 use Collective\Html\Eloquent\FormAccessible;
+use Grimzy\LaravelMysqlSpatial\Eloquent\SpatialTrait;
 use Illuminate\Database\Eloquent\Model;
 use Illuminate\Support\Arr;
+use App\Events\MemberUpdated;
+use App\Jobs\UpdateMemberGeolocation;
 
 class Member extends Model
 {
 
-	use FormAccessible;
+	use FormAccessible, SpatialTrait;
 
 	protected $guarded = ['id', 'created_at', 'updated_at'];
 
 	protected $dates = ['created_at', 'updated_at', 'geboortedatum'];
 
+	protected $spatialFields = ['geolocatie'];
+	
 	// Number of points needed for every level in the points system
 	const RANK_POINTS = [0, 3, 10, 20, 35, 50, 70, 100];
+	
+	protected $dispatchesEvents = [
+		'updated' => MemberUpdated::class
+	];
 
 	// Full name
 	public function getVolnaamAttribute()
 	{
 		return str_replace('  ', ' ', $this->voornaam . ' ' . $this->tussenvoegsel . ' ' . $this->achternaam);
 	}
 
+	// Full address as single string
+	public function getVolledigAdresAttribute()
+	{
+		return $this->adres . ', ' . $this->postcode . ' ' . $this->plaats;
+	}
+
+	public function getGeolocatieAttribute()
+	{
+		// Call geocoder if location is not yet available
+		if ($this->attributes['geolocatie'] === null) {
+			UpdateMemberGeolocation::dispatchSync($this);
+		}
+
+		$this->refresh();
+		return $this->attributes['geolocatie'];
```

Comment:
I don't think this is the best idea. What if a portion of the members, have a faulty adres. Every load of the map, will re-request all these faulty addresses and this wil go though our API calls. 

I would suggest to keep the error in the database, such that we know what went wrong the request of a single request and only re-request them every so often. With a command.

And of course if someone changes it's adress is should re-request it.

Reply:
You can see this already. follow the `laravel.log` and with the current test data open the map. You will see for-instance:
> Exception when geocoding address for member Jaap Jones: No geolocation match found for address 'Ringdijk Zuid 101, 4506 HD Cadzand'.